noir-projects/aztec-nr/aztec/src/note/note_getter.nr

```diff
@@ -146,14 +145,12 @@ where
     // to fail.
     let filter_fn = options.filter;
     let filter_args = options.filter_args;
-    let filtered_notes = filter_fn(opt_notes, filter_args);
+    let notes = filter_fn(retrieved_notes, filter_args);
 
-    let notes = array::collapse(filtered_notes);
     let mut note_hashes = BoundedVec::new();
 
-    // We have now collapsed the sparse array of Options into a BoundedVec. This is a more ergonomic type and also
-    // results in reduced gate counts when setting a limit value, since we guarantee that the limit is an upper bound
-    // for the runtime length, and can therefore have fewer loop iterations.
+    // This results in reduced gate counts when setting a limit value, since we guarantee that the limit is an upper
+    // bound for the runtime length, and can therefore have fewer loop iterations.
     assert(notes.len() <= options.limit, "Got more notes than limit.");
```

Comment:
This assert is what made me think about the limit in NoteGetter and why not have it be the max length. This code is now quite weird. It's hidden from the user though so it's probably ok.